api/net/Areas/Subscriber/Controllers/TonePoolController.cs

```diff
@@ -0,0 +1,118 @@
+using System.Net;
+using System.Net.Mime;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using TNO.API.Filters;
+using TNO.API.Models;
+using TNO.Core.Exceptions;
+using TNO.DAL.Services;
+using TNO.Entities;
+using TNO.Keycloak;
+using TonePoolModel = TNO.API.Areas.Subscriber.Models.TonePool.TonePoolModel;
+
+namespace TNO.API.Areas.Subscriber.Controllers;
+
+///<summary>
+///TonePoolController class, provides tone pool endpoints for the api.
+///</summary>
+///
+[ClientRoleAuthorize(ClientRole.Subscriber)]
+[ApiController]
+[Area("subscriber")]
+[ApiVersion("1.0")]
+[Route("api/v{version:apiVersion}/[area]/tonePool")]
+[Route("api/[area]/tonePool")]
+[Route("v{version:apiVersion}/[area]/tonePool")]
+[Route("[area]/tonePool")]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+
+public class TonePoolController: ControllerBase
+{
+    #region variables
+
+    private readonly ITonePoolService _service;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a TonePoolController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="service"></param>
+    ///
+    public TonePoolController(ITonePoolService service)
+    {
+        _service = service;
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Return an array of TonePool.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet, HttpHead]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<TonePoolModel>), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.NotModified)]
+    [SwaggerOperation(Tags = new[] { "TonePool" })]
+    [ETagCacheTableFilter("tone_pools")]
+    [ResponseCache(Duration = 5 * 60)]
```

Comment:
Have you tested this?

Reply:
I have not tested findbyId, just findByUserId. Going to test it.